src/XliffTasks/Model/XamlRuleDocument.cs

```diff
@@ -22,29 +24,88 @@ protected override IEnumerable<TranslatableNode> GetTranslatableNodes()
         {
             foreach (XElement? element in Document.Descendants())
             {
+                // first, let's check if the element has a descendent by the name of {elementLocalName.Description/DisplayName}
+                var descendentDisplayName = element.Descendants(XName.Get($"{element.Name.LocalName}.DisplayName", element.Name.NamespaceName)).FirstOrDefault();
+                var descendentDescription = element.Descendants(XName.Get($"{element.Name.LocalName}.Description", element.Name.NamespaceName)).FirstOrDefault();
```

Comment:
This should only be searching children, not all descendants.

```suggestion
                var descendentDisplayName = element.Elements(XName.Get($"{element.Name.LocalName}.DisplayName", element.Name.NamespaceName)).FirstOrDefault();
                var descendentDescription = element.Elements(XName.Get($"{element.Name.LocalName}.Description", element.Name.NamespaceName)).FirstOrDefault();
```